src/Matroids/matroids.jl

```diff
@@ -0,0 +1,322 @@
+	dual_matroid, direct_sum, restriction, deletion, contraction, minor,
+	principal_extension,
+	uniform_matroid, fano_matroid, non_fano_matroid
+
+################################################################################
+##  Constructing
+################################################################################
+
+struct Matroid
+    pm_matroid::Polymake.BigObject
+    groundset::Vector # groundset of the matroid 
+    gs2num::Dict{Any,Int}# dictionary to map the groundset to the integers from 1 to its size
+end
+
+pm_object(M::Matroid) = M.pm_matroid
+
+@doc Markdown.doc"""
+A matroid with bases `B` on the ground set ``{1,2,...`n`}`` (which can be the empty set) where `n` is a non-negative integer and `B` is a collection of subsets of the ground set satisfying a exchange property.
+
+# Examples
+To construct a rank two matroid with five bases on the four elements 1,2,3,4 you can write:
+```jldoctest
+julia> B = [[1,2],[1,3],[1,4],[2,3],[2,4]];
+
+julia> M = matroid_from_bases(B,4);
+A matroid on 4 elements with 5 bases.
+```
+
+# Examples
+To construct the same matroid on the four elements 2,1,i,j you may write:
+```jldoctest
+julia> M = matroid_from_bases([[1,2],[1,'i'],[1,'j'],[2,'i'],[2,'j']],[2,1,'i','j']);
+```
+"""
+matroid_from_bases(bases::Union{AbstractVector{<:AbstractVector{<:Base.Integer}}, AbstractVector{<:AbstractSet{<:Base.Integer}}}, nelements::Int) = matroid_from_bases(bases,Vector(1:nelements))
+
+function matroid_from_bases(bases::Union{AbstractVector{<:AbstractVector}, AbstractVector{<:AbstractSet}},groundset::AbstractVector, check::Bool=true)
+	if check_input && size(groundset)[1]!=length(Set(groundset))
+		throw("Input is not a valid groundset of a matroid")
+	end
+	gs2num = Dict{Any,Int}()
+	i = 1
+	for elem in groundset
+		gs2num[elem] = i
+		i+=1
+	end
+	pm_bases = [[gs2num[i]-1 for i in B] for B in bases]
+    	M = Polymake.matroid.Matroid(BASES=pm_bases,N_ELEMENTS=size(groundset)[1])
+	if check && !Polymake.matroid.check_basis_exchange_axiom(M.BASES)
+		throw("Input is not a collection of bases")
+	end
+    	return Matroid(M,groundset,gs2num)
+end
+
+@doc Markdown.doc"""
+A matroid with circuits `C` on the ground set ``{1,2,...`n`}`` (which can be the empty set) where `n` is a non-negative integer and `C` is a collection of subsets of the ground set.
+
+# Examples
+To construct a rank two matroid with five bases on the four elements 1,2,3,4 you can write:
+```jldoctest
+julia> B = [[1,2,3],[1,2,4],[3,4]];
+
+julia> M = matroid_from_circuits(B,4);
+A matroid on 4 elements with 5 bases.
+```
+# Examples
+To construct the same matroid on the four elements 2,1,i,j you may write:
+```jldoctest
+julia> M = matroid_from_bases([[1,2,'i'],[1,2,'j'],['i','j']],[2,1,'i','j']);
+```
+"""
+matroid_from_circuits(circuits::Union{AbstractVector{<:AbstractVector{<:Base.Integer}}, AbstractVector{<:AbstractSet{<:Base.Integer}}}, nelements::Int) = matroid_from_circuits(circuits,Vector(1:nelements))
```

Comment:
I recommend using our `IntegerUnion` type here, so that also `fmpz` is supported